src/configupdater/configupdater.py

```diff
@@ -289,16 +323,17 @@ def __str__(self):
             s += str(entry)
         return s
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return "<Section: {}>".format(self.name)
 
-    def __getitem__(self, key):
+    def __getitem__(self, key: str) -> "Option":
         key = self._container.optionxform(key)
-        if key not in self.options():
-            raise KeyError(key)
-        return self._structure[self._get_option_idx(key=key)]
+        try:
+            return next(o for o in self.iter_options() if o.key == key)
+        except StopIteration:
+            raise KeyError("No option `{}` found".format(key), {"key": key})
```

Comment:
Would it make sense to do
```
except StopIteration as e:
  raise KeyError("No option `{}` found".format(key), {"key": key}) from e
```
Using raise from helps when debugging as it keeps the stack trace.